packages/matter.js/src/cluster/server/GeneralCommissioningServer.ts

```diff
@@ -54,29 +54,32 @@ export const GeneralCommissioningClusterHandler: (options?: {
                     debugText: `Country code change not allowed: ${countryCode}`
                 };
             }
-            basicInformationCluster.setLocationAttribute(countryCode);
+            if (countryCode !== "XX") {
+                basicInformationCluster.setLocationAttribute(countryCode);
+            }
         }
 
         // Check and handle regulatory config for LocationCapability
-        let validValues;
-        switch (locationCapabilityValue) {
-            case (RegulatoryLocationType.Outdoor):
-                validValues = [RegulatoryLocationType.Outdoor];
-                break;
-            case (RegulatoryLocationType.Indoor):
-                validValues = [RegulatoryLocationType.Indoor];
-                break;
-            case (RegulatoryLocationType.IndoorOutdoor):
-                validValues = [RegulatoryLocationType.Indoor, RegulatoryLocationType.Outdoor];
-                break;
-            default:
+        if (newRegulatoryConfig !== RegulatoryLocationType.IndoorOutdoor) { // Ignore setting IndoorOutdoor
+            let validValues;
+            switch (locationCapabilityValue) {
+                case (RegulatoryLocationType.Outdoor):
+                    validValues = [RegulatoryLocationType.Outdoor];
+                    break;
+                case (RegulatoryLocationType.Indoor):
+                    validValues = [RegulatoryLocationType.Indoor];
+                    break;
+                case (RegulatoryLocationType.IndoorOutdoor):
+                    validValues = [RegulatoryLocationType.Indoor, RegulatoryLocationType.Outdoor];
```

Comment:
I can't find mention in the specification where it says Indoor/Outdoor is illegal for RegulatoryConfig. In fact my reading is the default would be Indoor/Outdoor if LocationCapability is Indoor/Outdoor. Maybe we should accept the value?

Reply:
I checkChip tool again and yes it is accepted (in fact makes no sense, but) :-) I adjust